### PR TITLE
refactor: adjust account discovery logic

### PR DESCRIPTION
**Pull request by CedrikNikita**
https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#address-gap-limit

### CODE REVIEW

File: src/store/modules/accounts/hdWallet.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ import {
   isTxOfASupportedType,
 } from '../../../popup/utils';
 
+const ADDRESS_GAP_LIMIT = 5;
```
Comment (peronczyk) [inline src/store/modules/accounts/hdWallet.js:18]: Can you add short comment about what this is?

**Review by peronczyk — APPROVED** [review]

File: src/store/modules/accounts/hdWallet.js [hunk 2]
```diff
@@ -33,11 +40,31 @@ export default {
...
       let lastNotEmptyIdx = 0;
-      let account;
-      for (let nextIdx = state.nextAccountIdx; nextIdx <= 5; nextIdx += 1) {
-        account = getHdWalletAccount(rootGetters.wallet, nextIdx);
-        // eslint-disable-next-line no-await-in-loop
-        if (await dispatch('isAccountUsed', account.address)) lastNotEmptyIdx = nextIdx;
+      let lastIndex = 0;
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```
Comment (subhod-i) [inline src/store/modules/accounts/hdWallet.js:46]: Infinite loops must be avoided. I feel this logic is not easy to understand. Maybe you can simplify or add comments

File: src/store/modules/accounts/hdWallet.js [hunk 3]
```diff
@@ -33,11 +40,31 @@ export default {
...
+            Array(ADDRESS_GAP_LIMIT + lastNotEmptyIdx - lastIndex + 1)
+              // eslint-disable-next-line no-loop-func
+              .fill().map((x, i) => i + lastIndex).map((index) => dispatch(
+                'isAccountUsed',
+                getHdWalletAccount(rootGetters.wallet, index).address,
+              )),
+          );
+          if (
+            isAccountUsedArray.lastIndexOf(true) === -1
+            || isAccountUsedArray.filter((x) => !x).length === ADDRESS_GAP_LIMIT
```
Comment (pstachel) [inline src/store/modules/accounts/hdWallet.js:59]: I'd suggest to stick to more meaningful variable names than `x`

**Review by pstachel — APPROVED** [review]

**Comment by Liubov-crypto** [comment]
LGTM

**Pull request merged**